packages/veto-app/frontend/src/hooks/useRequest.ts

```diff
@@ -0,0 +1,43 @@
+import useSWR, { ConfigInterface, responseInterface } from 'swr'
+import axios, { AxiosRequestConfig, AxiosResponse, AxiosError } from 'axios'
+
+export type GetRequest = AxiosRequestConfig
+
+interface Return<Data, Error>
+  extends Pick<responseInterface<AxiosResponse<Data>, AxiosError<Error>>, 'isValidating' | 'revalidate' | 'error'> {
+  data: Data | undefined
+  response: AxiosResponse<Data> | undefined
+}
+
+export interface Config<Data = unknown, Error = unknown>
+  extends Omit<ConfigInterface<AxiosResponse<Data>, AxiosError<Error>>, 'initialData'> {
+  initialData?: Data
+}
+
+export default function useRequest<Data = any, Error = any>(
```

Comment:
`Data` & `Error` follow types from `swr` but having `any` means we need to be careful to not overlook our types when calling useRequest.

For the sake of rapid iteration and prototyping is ok to keep `any` here but we should add types in most cases before integrating the code with the main branch.

Reply:
Also, in our case, can `Error` be anything else than a JS error?